### PR TITLE
Add misc code tests to be executed by CI

### PR DESCRIPTION
**Pull request by Myoldmopar**
This starts a pattern for adding tests to be run by CI from Python scripts.  The first example is verifying that the testfiles/CMakeLists.txt file actually includes all the idfs and imfs in there.  Which, now, I see it currently _is not_.  So the script has already been made useful.

There still needs some refinement.  Previously, we only required devs to have Python if they were doing specifically regression testing or if they were building a package.  For right now, I have changed it so that it goes for ANY testing.  This is likely too far.  But the only other option I can think of is to add another CMAKE flag, like ENABLE_PYTHON_TESTS.  

@kbenne @mbadams5 I'm curious your thoughts.  There are a bunch of tests that I have in mind for CI to run, so I want to make sure the pattern is set nicely up front.

### CODE REVIEW

File: scripts/dev/VerifyIDFsInCMake.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import sys
+import os
+import glob
```
Comment (Myoldmopar) [inline scripts/dev/VerifyIDFsInCMake.py:3]: Remove glob import

**Comment by Myoldmopar** [comment]
So, with 79623c9, the expand objects errors should go away.  On Linux and Mac, CI should be 100% green.  On Windows, there will still be a failure.  However, it should be a warning now.  I'm cleaning up that warning with my next commit, I just want to see if the warning is actually processed correctly before I fix it.

**Comment by Myoldmopar** [comment]
Woo!  Warning!  OK, next commit will be back to all green and we can discuss the remaining changes to get it acceptable for merge.

**Comment by Myoldmopar** [comment]
:fire: I added a CMake option, off by default, for enabling Python-based tests to run locally.  This goes along with my hope that we don't require any developers to have Python installed.  CI will happily run the Python-related stuff for them, as it already does for regressions. :fire:

**Comment by Myoldmopar** [comment]
All green, as expected.  OK, now does anyone object to my additions here?  I am adding a new CMake variable called BUILD_PYTHON_TESTS that is OFF by default.  If it is ON, it finds Python at configure time, and adds a new test called VerifyIDFsInCMakeFolder that will give warnings on the dashboard if an idf or imf file is found yet not listed in the testfiles/CMakeLists.txt file.  Any issues with this @kbenne or @mbadams5 or @mjwitte?

**Comment by mjwitte** [comment]
@Myoldmopar Seems fine to me, but the variable name "BUILD_PYTHON_TESTS" seems a bit misleading.  We've already got other options that depend on python (ENABLE_REGRESSION_TESTING and BUILD_PACKAGE) so can this name be more descriptive of what it's enabling?

**Comment by Myoldmopar** [comment]
That's a good point.  I'm not sure I have a solution right now.  This branch basically adds **miscellaneous** Python-based tests.  As you mentioned, there are other switches that end up requiring Python.  These are handled in this block [here](https://github.com/NREL/EnergyPlus/pull/5279/files#diff-af3b638bc2a3e6c650974192a53c7291R69).  In any case, if you enable any of these, then CMake will re-configure.  If you end up not having Python, you will get a warning right away.  If it finds Python then you are good-to-go.  

One option would be to only expose the BUILD_PYTHON_TESTS if BUILD_TESTING has been turned on.  That's the behavior of the BUILD_REGRESSION or whatever.   That doesn't really help the situation though.  Just doesn't expose it for normal executable-only builds.

Maybe there _is_ a renaming that would actually work.  BUILD_MISC_TESTS is dumb.  ENABLE_STATIC_TESTING?  If we can get away with just a simple rename, the Python dependency will still be described in the Cmake [declaration](https://github.com/NREL/EnergyPlus/pull/5279/files#diff-af3b638bc2a3e6c650974192a53c7291R49), and CMake will happily tell you to if it needs Python and can't find it at configure time.

**Comment by Myoldmopar** [comment]
CMake gurus unite!  I mean... @kbenne can you comment on this including the variable naming and just how I added the test into CMake?  Otherwise I'll be forced to make a fake github account called cmakeguru and comment on here about how great it looks.

**Comment by nrel-bot-3** [comment]
@Myoldmopar @lgentile it has been 14 days since this pull request was last updated.

**Comment by kbenne** [comment]
My first thought is to just bite the bullet and require Python as a basic dependency. You jumped on this train and it is going to be hard to stop. Adding a bunch of options fragments that project in my opinion.

**Comment by kbenne** [comment]
Once you have introduced Python as a core dependency you are about one tick away from Python HVAC components and Python control logic.

**Comment by Myoldmopar** [comment]
Did I just hear @kbenne volunteering to rewrite EMS in Python ..... ?

**Comment by Myoldmopar** [comment]
Hmmmm, OK.  Adding Python as a basic dependency is likely not a big problem.  The whole dev team falls into at least one of these:
- uses Mac OS X, so already has Python
- has run the E+ test suite or other tools previously and has Python installed
- is capable of installing Python **very** easily

@mjwitte What are your thoughts on this?  Right now, the only core dependencies are CMake and a C++ compiler in order to build the EnergyPlus.exe binary.  Adding Python as a core dependency is actually a pretty lightweight one, and it would make it clear that we want to use Python for many utility functions moving forward, and could actually make it cleaner moving forward by keeping other languages from creeping in.  And it would make the CMake configuration simpler as @kbenne mentioned.

**Comment by mjwitte** [comment]
@Myoldmopar @kbenne I'm not following what we gain by making Python a core dependency.   A few less options in CMake? Either way, we still plan to make these additional tests optional, yes?  Most active developers won't care, because they probably have Python installed anyway.  Will the average casual developer out there care?  The one who _simply_ wants to build a copy of EnergyPlus and maybe make some minor modifications.

**Comment by jasondegraw** [comment]
@Myoldmopar Yes, but what version of Python? I haven't used my favorite emoji in at least a week...

**Comment by Myoldmopar** [comment]
@jasondegraw Favorite Emoji?

One of these is my favorite, but I can't decide...
:feelsgood: :finnadie: :goberserk: :godmode: :hurtrealbad: :rage1: :rage2: :rage3: :rage4: :suspect:

**Comment by kbenne** [comment]
I just resist creating "classes" of developers. Those that can build some targets but not others. Some developers who run some tests but not others, etc. Plus configuration options tend to proliferate through the project over time. It doesn't seem so bad at first but gradually it will get more complicated. 

I wasn't thrilled about Python as a dependency when it happened, but there were bona fide reasons for bringing it in, so now all I'm saying is own it. Commit to that decision.

**Comment by Myoldmopar** [comment]
Hmmmm, another option, @kbenne, is to not have it exposed as an option at all, but still leave it conditional in CMake.  Basically if the finder finds Python, enable some extra tests to be run and all that.  If it doesn't, don't enable it.  So optional things that enable enhanced capability; feels like `./configure` style.....

**Comment by mjwitte** [comment]
@Myoldmopar I don't really have a strong opinion, it's just one more install.  Just add it to the appropriate places in the main readme.  That said, I contend there will always be two (or more) classes of developers - those who want to contribute real code and those who just want to build for their own purposes.  And we still have other targets that need other tools that are optional.

**Comment by kbenne** [comment]
Yeah it isn't that big of a deal either way. I will say we watched Ruby creep into OpenStudio like this and after all the effort put in  conditionals if Ruby, ultimately it became so fundamental we ripped it all out. 

I suspect it will happen here to. Somebody someday is going to make Python components a thing.

**Comment by Myoldmopar** [comment]
Did I just hear @kbenne say he's already implemented Python components ..... ?

**Comment by macumber** [comment]
+1 for minimizing configuration options and +2 for python components, @kbenne is on a roll :-)  If someone defines and implements the E+ API I'll volunteer to write the interpreter ;-)

**Comment by nrel-bot-3** [comment]
@Myoldmopar @lgentile it has been 14 days since this pull request was last updated.

**Comment by Myoldmopar** [comment]
Alright, this is now more ready to go.  As directed through discussion, this change make Python a required part of our development project.  Right now it is looking for 2.7+.  If we decide to require 3.0+ we can make that change later.

**Comment by Myoldmopar** [comment]
Anyone have an objection to this merging?  Anyone devs listening out there that don't have Python installed?

**Comment by Myoldmopar** [comment]
@mjwitte I'm in a reviewy and mergy mood today.  I don't think there is anything else lingering for people to discuss here, so I'm likely to push the green button myself if the clock hits 10 and no one has convinced me to not push said green button... :clock3: :clock2: :clock1:

**Comment by mjwitte** [comment]
It always makes me a little nervous when you get mergy . . .
No objections here, been using Python this entire century.

**Comment by mjwitte** [comment]
And just for proper etiquette, I'll merge it for you.

**Comment by Myoldmopar** [comment]
> It always makes me a little nervous when you get mergy . . .

~~Hey, at least we're not just unlocking files and letting people push code in without seeing it first! :wink:.~~ @myoldmopar WOULD LIKE TO RECALL THIS MESSAGE

**Pull request merged**